components/NameTagForm.tsx

```diff
@@ -54,24 +58,33 @@ export function NameTagForm({
             <option value="other">Other</option>
           </select>
         </div>
-        <div>
-          <label>Self Disclosure</label>
+        <div style={{ paddingBottom: bottom_padding }}>
+          <label>Something About Me</label>
           <input
             className="text-input"
-            defaultValue={content.disclosure}
-            {...register("disclosure")}
+            defaultValue={content.disclosure || "I have a stutter"}
+            {...register("disclosure", { maxLength: maxDisclosureLength })}
           />
+          <div className={`char-count ${isOverLimit ? 'warning' : ''}`}> 
+            <span>
+              {disclosureValue.length}/{maxDisclosureLength}
+            </span>
+            <span className="char-limit-info">
+              (Maximum characters allowed)
+            </span>
+            {isOverLimit && <span className="warning-message">Exceeded length limit!</span>}
+          </div>
         </div>
         <div className="form-container">
           <div className="controller-container">
             <Controller
               control={control}
               name="visible"
-              defaultValue={content.visible}
+              defaultValue={false}
               render={({ field: { onChange, value } }) => (
                 <FormControlLabel
                   control={
-                    <Checkbox checked={value} onChange={onChange} />
+                    <Switch checked={value} onChange={(e) => { onChange(e); handleSubmit(onNameTagContentChange)();}} type="checkbox"/>
```

Comment:
NameTag content won't be saved to DB without displaying the nametag. Consider adding a "save" button to update DB.

Reply:
```onNameTagContentChange()``` currently save the nametag content in DB as well. But since we have a separate "Save" button now, we should avoid unnecessary DB write. Consider refactor ```onNameTagContentChange()``` to only display/hide the nametag based on the switch status.